packages/app-builder/src/services/editor/ast-editor.tsx

```diff
@@ -315,16 +317,56 @@ export function useRootOrAndChildValidation(stringPath: string) {
   ]);
 }
 
-export function useValidationStatus(stringPath: string) {
+export function useValidationStatus(
+  stringPath: string,
+  returnValue?: ReturnValue,
+) {
   const path = usePath(stringPath);
   const evaluationErrors = useEvaluationErrors(stringPath);
   const parentPath = useParentPath(path);
+  const parentNode = useParentAstNode(parentPath);
+  const isDivByZeroField =
+    isDivisionDenominator(parentNode, path) &&
+    !returnValue?.isOmitted &&
+    returnValue?.value === 0;
+
   const parentEvaluationErrors = useParentEvaluationErrors(parentPath);
+  const valueIsNull = !returnValue?.isOmitted && returnValue?.value === null;
   return React.useMemo(() => {
-    return getValidationStatus(
+    return getValidationStatus({
       evaluationErrors,
+      valueIsNull,
+      isDivByZeroField,
       parentEvaluationErrors,
-      parentPath?.childPathSegment,
-    );
-  }, [evaluationErrors, parentEvaluationErrors, parentPath]);
+      pathSegment: parentPath?.childPathSegment,
+    });
+  }, [
+    evaluationErrors,
+    isDivByZeroField,
+    parentEvaluationErrors,
+    parentPath?.childPathSegment,
+    valueIsNull,
+  ]);
+}
+
+function isDivisionDenominator(
+  parentNode: Tree<AstNode> | undefined,
+  path: Path,
+) {
+  if (path.length === 0) {
```

Comment:
or would I be correct to say that this part is redundant with line 359 below ? (My brain just has difficulty to write indexes that possibly don't exist 🙈 )

Reply:
Yes, you can ommit it safelly. Index access return `undefined` (and do not throw) when this is out of bound in JS.

But you can keep this, it's not a problem

Reply:
There is a "new" [at](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/at) array accessor that make code like this easier to read IMO (because it's not using the access syntax used in a lot of programming language w\ the same conventions)